src/fastfeedparser/main.py

```diff
@@ -44,640 +47,573 @@ def parse(source):
-                for ns, uri in namespaces.items():
-                    if uri == "http://purl.org/rss/1.0/modules/content/":
-                        content = item.find(f"{{{uri}}}encoded")
-                        break
-            if content is None:
-                content = item.find("content")
-        elif feed_type == "atom":
-            content = item.find("{http://www.w3.org/2005/Atom}content")
-
-        if content is not None:
-            content_type = content.get("type", "text/html")  # Default to text/html
-            if content_type in ["html", "xhtml"]:
-                # For XHTML content, serialize the entire content
-                content_value = etree.tostring(
-                    content, encoding="unicode", method="xml"
-                )
-            else:
-                content_value = content.text if content.text else ""
-
+    # If content is still empty, try to use description
+    if "content" not in entry:
+        description = item.find("description")
+        if description is not None and description.text:
             entry["content"] = [
                 {
-                    "type": content_type,
-                    "language": content.get(
-                        "{http://www.w3.org/XML/1998/namespace}lang"
-                    ),
-                    "base": content.get("{http://www.w3.org/XML/1998/namespace}base"),
-                    "value": content_value,
-                }
+                    "type": "text/html",
+                    "language": item.get("{http://www.w3.org/XML/1998/namespace}lang"),
+                    "base": item.get("{http://www.w3.org/XML/1998/namespace}base"),
+                    "value": description.text,
+                },
             ]
 
-        # If content is still empty, try to use description
-        if "content" not in entry or not entry["content"]:
-            description = item.find("description")
-            if description is not None and description.text:
-                entry["content"] = [{"type": "text/html", "value": description.text}]
-
-        if ("description" not in entry) and ("content" in entry or "summary" in entry):
-            content = (
-                entry.get("content", [{}])[0].get("value", "")
-                if entry.get("content")
-                else ""
-            )
-            if content:
+    # If description is empty, derive it from content (removing non-text content)
+    if "description" not in entry and "content" in entry:
+        content = entry["content"][0]["value"]
+        if content:
+            try:
+                html_content = etree.HTML(content)
+                if html_content is not None:
+                    content_text = html_content.xpath("string()")
+                    if isinstance(content_text, str):
+                        content = re.sub(r"\s+", " ", content_text)
+            except etree.ParserError:
+                pass
+        entry["description"] = content[:512]
```

Comment:
BREAKING: html_content was previously unrestricted in length, 512 limit is now always used.